pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -415,9 +417,12 @@ func (m *kubeGenericRuntimeManager) getPodContainerStatuses(uid kubetypes.UID, n
 					tMessage = m.readLastStringFromContainerLogs(status.GetLogPath())
 				}
 			}
-			// Use the termination message written by the application is not empty
+			// Enrich the termination message written by the application is not empty
 			if len(tMessage) != 0 {
-				cStatus.Message = tMessage
+				if len(cStatus.Message) != 0 {
```

Comment:
Can you explain a little more about this change? I'm not totally following how it relates to the overall change you are describing in your pr.

Reply:
Thanks for your review.
You are right, strictly speaking, it should not belong to this PR, however, in more general sense, it can prevent future useful info being lost.
For example, kubelet failed to read the termination file which may due to a reason described in dockerd's container error message. 
**So the principle here is that, discard exit info ONLY if we can ensure it is useless or duplicated**.
For this case, we cannot ensure that dockerd's container error message is useless given "failed to read the termination file" or even we successfully read the file (because we also donot know if the dockerd's container error message is the root cause of the just read content).

So, for simplicity, I just append the error message, instead of one replace another.

It is just my preference, and TBD, we can disscus on it. :)

(Release note also updated)